elfi/model/elfi_model.py

```diff
@@ -458,33 +461,52 @@ def _give_name(self, name, model):
                 name = self._new_name(name[:-1], model)
             return name
 
+        try:
+            name = self._inspect_name()
+        except:
+            logger.warning("Automatic name inspection failed, using a random name "
+                           "instead. This may be caused by using an interactive Python "
+                           "shell. You can provide a name parameter e.g. "
+                           "elfi.Prior('uniform', name='nodename') to suppress this "
+                           "warning.")
+            name = None
+
+        if name is None or model.has_node(name):
+            name = self._new_name(model=model)
+
+        return name
+
```

Comment:
I'm not convinced this should be default behaviour. When running in batch mode (e.g on a cluster) you will want to have as little magic (read hard to debug behaviour) going on as possible. Where this would be useful is for quick prototyping in interactive mode. So maybe there should be an optional setting to enable automatic naming.